src/client/runner/automation/playback/scroll.js

```diff
@@ -1,364 +1,185 @@
+var Promise        = hammerhead.Promise;
 var messageSandbox = hammerhead.eventSandbox.message;
-
-var CROSS_DOMAIN_MESSAGES = testCafeCore.CROSS_DOMAIN_MESSAGES;
-var domUtils              = testCafeCore.domUtils;
-var positionUtils         = testCafeCore.positionUtils;
-var styleUtils            = testCafeCore.styleUtils;
+var domUtils       = testCafeCore.domUtils;
+var styleUtils     = testCafeCore.styleUtils;
+var positionUtils  = testCafeCore.positionUtils;
 
 
 const MAX_SCROLL_MARGIN = 50;
-const MIN_SCROLL_STEP   = 10;
-const SCROLL_SPEED      = 1;
-const SCROLL_DELAY      = 10;
-
-
-function getScrollOption (currentScrollable, target, offsetX, offsetY) {
-    var targetDimensions     = positionUtils.getClientDimensions(target),
-        scrollableDimensions = positionUtils.getClientDimensions(currentScrollable),
-
-        scrollTop            = null,
-        scrollLeft           = null,
-        maxLeftScrollMargin  = Math.min(MAX_SCROLL_MARGIN, Math.floor(scrollableDimensions.width / 2)),
-        maxTopScrollMargin   = Math.min(MAX_SCROLL_MARGIN, Math.floor(scrollableDimensions.height / 2)),
-
-        isScrollDown         = -1,
-        isScrollRight        = -1;
-
-    var relation = {
-        top:    targetDimensions.top - (scrollableDimensions.top + scrollableDimensions.border.top),
-        bottom: scrollableDimensions.bottom - scrollableDimensions.border.bottom -
-                scrollableDimensions.scrollbar.bottom - targetDimensions.bottom,
-        left:   targetDimensions.left - (scrollableDimensions.left + scrollableDimensions.border.left),
-        right:  scrollableDimensions.right - scrollableDimensions.border.right - scrollableDimensions.scrollbar.right -
-                targetDimensions.right
-    };
-
-    // vertical scroll
-    if (relation.top < 0)
-        scrollTop = Math.round(scrollableDimensions.scroll.top + relation.top - maxTopScrollMargin);
-    else if (relation.top > 0 && relation.bottom < 0) {
-        scrollTop    = Math.round(scrollableDimensions.scroll.top + Math.min(relation.top, -relation.bottom) +
-                                  maxTopScrollMargin);
-        isScrollDown = 1;
-    }
 
-    // horizontal scroll
-    if (relation.left < 0)
-        scrollLeft = Math.round(scrollableDimensions.scroll.left + relation.left - maxLeftScrollMargin);
-    else if (relation.left > 0 && relation.right < 0) {
-        scrollLeft    = Math.round(scrollableDimensions.scroll.left + Math.min(relation.left, -relation.right) +
-                                   maxLeftScrollMargin);
-        isScrollRight = 1;
-    }
 
-    //NOTE: we should check: can we show the full element
-    var targetElementWidth  = Math.abs(targetDimensions.right - targetDimensions.left),
-        targetElementHeight = Math.abs(targetDimensions.bottom - targetDimensions.top);
+// Setup cross-iframe interaction
+messageSandbox.on(messageSandbox.SERVICE_MSG_RECEIVED_EVENT, e => {
+    if (e.message.cmd === 'scroll') {
```

Comment:
Maybe we should use constant for this message `cmd`? And shouldn't it be more specific to avoid collisions, e.g. `automation|scroll` or smthg